ops/pebble.py

```diff
@@ -1135,6 +1135,64 @@ def __eq__(self, other: Union['CheckDict', 'Check']) -> bool:
         else:
             return NotImplemented
 
+    def _merge_exec(self, other: 'ExecDict') -> None:
+        """Merges this exec object with another exec definition.
+
+        For attributes present in both objects, the passed in exec
+        attributes take precedence.
+        """
+        for name, value in other.__dict__.items():
+            if not value:
+                continue
+            if name == 'environment':
+                getattr(self.exec, name).update(value)
+            else:
+                setattr(self.exec, name, value)
```

Comment:
So I think this is close, and the `.update()` is a good idea. However, there are a few issues here:

- `other` is a dict here, so it should just be `other.items()`, not `other.__dict__.items()`
- because it's a dict, we can't use `getattr` and `setattr`, `self.exec[name]` and `self.exec[name] = value` would work
- if the `environment` key isn't present in the current `self.exec` dict, we need to handle that case -- probably with `self.exec.get()`
- unfortunately you can't directly update, you have to reassign, for the case when it's using the `get` default
- I think we probably want `if self.exec is None: self.exec = {}` at the start of these methods, in case the base layer doesn't have it; seems unlikely to happen in practice, but still a good safeguard

Overall, I think we want something more like this:

```python
        if self.exec is None:
            self.exec = {}
        for name, value in other.items():
            if not value:
                continue
            if name == 'environment':
                current = self.exec.get(name, {})
                current.update(value)
                self.exec[name] = current
            else:
                self.exec[name] = value
```

Similar with `_merge_http` and `_merge_tcp`.

We should probably add tests of these cases too, to the merge test.